src/Escaper/Escaper.h

```diff
@@ -0,0 +1,111 @@
+/**@file
+ * This file is part of the TASTE Runtime Common.
+ *
+ * @copyright 2021 N7 Space Sp. z o.o.
+ *
+ * TASTE Runtime Common was developed under a programme of,
+ * and funded by, the European Space Agency (the "ESA").
+ *
+ * Licensed under the ESA Public License (ESA-PL) Permissive,
+ * Version 2.3 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://essr.esa.int/license/list
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ESCAPER_H
+#define ESCAPER_H
+
+/**
+ * @file    Escaper.h
+ * @brief   Main header file for escaping mode
+ *
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stddef.h>
+
+#include "EscaperInternal.h"
+
+/// @brief Enumeration listing possible parser states
+typedef enum
+{
+    Escaper_State_Wait,
+    Escaper_State_Data_Byte,
+    Escaper_State_Escape_Byte,
+} Escaper_PacketParseState;
+
+/// @brief Structure representing escaper
+typedef struct
+{
+    Escaper_PacketParseState m_parse_state;
+    bool m_encode_started;
+    bool m_escape;
+    bool m_encode_finished;
+    uint8_t* m_encoded_packet_buffer;
```

Comment:
Maybe you should create a function `Escaper_init`, which shall be used to initialize values of pointers to buffers and their sizes? It would be nice.

Reply:
Finished